### PR TITLE
Milestonethree

### PR DESCRIPTION
**Pull request by cgcarlosg**
The method saves, in the friendship model, creates two rows
using a call -back, after_save to create a second record after saving a friendship (using user_id and friend_id in inverse order)
user_id and friend_id is unique

### CODE REVIEW

**Review by MarilenaRoque — CHANGES_REQUESTED** [review]
Hello @cgcarlosg! :rocket:

## Congratulations on your work on this milestone, so far! Almost all features are implemented smoothly! 👏🏻 

However, there are a few details to adjust before the approval.

# Changes Requested :recycle:

- [ ] Posts timeline includes posts written by logged-in user and their friends.
- Remember that we have a model of mutual friendship. It means that once a friendship is confirmed, no matter who sent the invitation, both should see each other posts in their timelines. Right now, only the one that sent the invitation (the one represented in the column user_id in the friendship) can see the post of his friends. Make sure to combine queries to get the friend's posts where the current user is user_id and also where he is the friend_id.

- [ ] Unit and Integration tests!
- Please, add more unit and integration tests, including all your models.

- [ ] Fix misspelled command to run your tests on README.md

That is all. Feel free to contact me on Slack if you need any further information.

Happy Coding!!! 💻 

**_Marilena Roque
Full-Stack Developer
Code Reviewer at Microverse Inc.
[Portfolio](https://mariroqueproject.com/) | [Linkedin](https://www.linkedin.com/in/roquemarilena/) 
marilenas@id.uf.br_**

File: spec/friendrequest_spec.rb [hunk 1]
```diff
@@ -21,3 +21,37 @@
...
+      expect(f1.valid?).to eq(true)
+    end
+
+    it 'create valid friendship' do
+      f1 = Friendship.new(user_id: @a.id, friend_id: @b.id, confirmed: true)
+      f1.save
+      expect(Friendship.all.length).to eq(2)
+    end
+
+    it 'create invalid friendship' do
```
Comment (ghost) [inline spec/friendrequest_spec.rb]: You cannot write tests for multiple classes in one file. `User` class tests should be written in `user_spec.rb` and `Friendship` class tests in `friendship_spec.rb`.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required 🛑 

## Hello team 👋,

**Great job implementing the requested changes 👏. However, there are still a few requirements needed.**

1).**So I have two users, Jen and Mercy. When Jen clicks the add friend button next to Mercy's name, then after the button is removed there should be a sentence reading pending status or something, next to Mercy's name. After they are confirmed friends, it should read Friends next to their names. See an example in the screenshot below.**

 before a request is sent  | After a request is sent
 ---------------------------| ------------------------
before_request | after_request**

2). **You have not used the correct directory structure for the spec tests. Model tests should be in `spec/models` directory and feature tests should be `spec/features` directory.**

3). **Add tests for associations and methods defined in all your models. Hint you can use [Shoulda Matchers](http://matchers.shoulda.io/docs/v4.0.1/Shoulda/Matchers/ActiveRecord.html#belong_to-instance_methodl). If a method is not used outside a class, you can declare it private and not write a test for it.**
4). **Add feature tests for when you create a post, comment on a post and like a post.**
5). **See on more in the comment below.**

## Happy coding 🥂.

**You can reach me on slack (Rose Wanjohi).**

**Review by MarilenaRoque — APPROVED** [review]
## Congratulations !!! 🎉 

# Project Approved! ✔️ 

You can merge it, and go on with the next milestone!!!

As always, Happy Coding! 💻 

**_Marilena Roque
Full-Stack Developer
Code Reviewer at Microverse Inc.
[Portfolio](https://mariroqueproject.com/) | [Linkedin](https://www.linkedin.com/in/roquemarilena/) 
marilenas@id.uf.br_**

**Pull request merged**